### ISSUE TITLE
1.1.0

### ISSUE THREAD

**Issue by k00ni** [easyrdf/easyrdf#365]
I would like to create a new minor release to ship our latest work.

https://github.com/easyrdf/easyrdf/releases/edit/untagged-aa30be1dad1c73a5e8b3 (only for collaborators)

Content so far:
```
Bug fix release with a few improvements and changes in Composer dependencies.

**Highlights:**
* Composer: 
  * changed ARC2 from `~2.2` to `^2.4` (#347)
  * replaced `sami/sami` with `code-lts/doctum`
  * added the following required PHP extensions: ext-pcre, ext-xmlreader, lib-libxml
* Accept an array of properties in get/getLiteral/getResource methods, fixes #325 (#326)
* Add support for parsing large SPARQL XML Results (#357)

**Highlights for EasyRdf developers:**
* added .editorconfig (#340)
* added a basic Docker setup to improve local development and testing (#334)
* added `CONTRIBUTING.md`
```

As far as I can see there are a few things to do before the release:

* [x] adapt milestones
* ~~based on `composer.json` this lib requires "ext-mbstring" ~~which isn't part of PHP 8 anymore (and might cause errors)~~ We could replace it with `symfony/polyfill-mbstring` to allow support for all available PHP versions (7.1+)~~

What do you think?

CC @njh @osma @zozlak

**Comment by zozlak** [comment]
Are you sure mbstring will be dropped? I can't see anything about it on https://github.com/php/php-src/blob/php-8.0.0rc1/UPGRADING. In fact it contains quite some info on changes in the extension API but definitely not about its removal.

**Comment by k00ni** [comment]
It seems like in PHP 8 some functions of `mbstring` will be changed or removed entirely. @Connum from "smalot/pdfparser" had problems with `mb_convert_encoding` in PHP 8 (ref: https://github.com/smalot/pdfparser/issues/335#issue-702564725). We fixed it by adding the polyfill I referenced.

Based on PHP documentation `mb_convert_encoding` is only available in PHP 7.x (source: https://www.php.net/manual/en/function.mb-convert-encoding.php). But maybe its outdated.

Furthermore, I am not even sure if we use functions of this extension (still). A quick search showed that we aren't using `mb_convert_encoding`.

**Comment by zozlak** [comment]
As PHP 8 hasn't been released yet, the official documentation simply doesn't mention PHP 8 yet (compare https://www.php.net/manual/en/function.substr.php - it's also listed as `(PHP 4, PHP 5, PHP 7)` but for sure they don't remove it in PHP 8) :-)

I wouldn't change to polyfill until it's required (until we really use one of removed functions) because this will for sure hurt the performance (and performant `mb_substr()` is crucial for parsers performance).

**Comment by njh** [comment]
Getting a 1.1.0 release out sounds like a good idea.

#350 / #352 is nearly finished - so would really like to include that.

What do you think about the other issues in https://github.com/easyrdf/easyrdf/milestone/11 ?
Should they be moved to a future milestone?

**Comment by k00ni** [comment]
I would move #265 and #314 to a future release unless someone wants to jumps in and help with a PR.

Error in #366 seems weird, but I didn't investigate it yet.

Could #350 benefit from your latest work on the SPARQL result parser?

**Comment by k00ni** [comment]
Hi. Because of the current situation I would prefer if we release what we have and move open issues to 1.2.0. Any objections @njh?

**Comment by k00ni** [comment]
If there are no objections until end of this week, I will release what we have as `1.2.0`.

**Comment by k00ni** [comment]
@njh can you update your website etc. to show the latest release?

**Comment by njh** [comment]
Website updated to use EasyRdf version 1.1.0 👍
https://github.com/easyrdf/www.easyrdf.org/commit/60facf30bb883480067f930465a5140f75489229

Would be good to get another release out very soon with the large RDF/XML file fix in it - maybe 1.1.1?
